internal/plans/retrieve.go

```diff
@@ -43,7 +44,11 @@ func (c *Client) Retrieve() *cobra.Command {
 			filters := c.Servicer.Filters()
 
 			if filters["type"] != "" {
-				request = request.Type_(filters["type"])
+				validType, err := metal.NewFindPlansTypeParameterFromValue(filters["type"])
+				if err != nil {
+					return err
```

Comment:
I opted to just return the error here rather than wrapping it, since the error that's thrown is already helpful: https://github.com/equinix-labs/metal-go/blob/main/metal/v1/model_plan_type.go#L60